### ISSUE TITLE
ODK-X Services Server Settings Server Password Visibility Icon Is Incorrect  

### ISSUE THREAD

**Issue by clarlars** [opendatakit/tool-suite-X#257]
When a user enters the Server Password in Server Settings, the visibility icon shows the password when visibility is hidden and hides the password otherwise (see attached image).  It should be the other way around.  This happens in ODK-X Services 2.1.8 rev 242p.  
![password_visibility_icon](https://user-images.githubusercontent.com/8062768/115100024-bae66b00-9eee-11eb-8496-b604c79c2fdc.jpg)

**Comment by sarahelshabrawy** [comment]
I can work on it

**Comment by wbrunette** [comment]
@sarahelshabrawy I have assigned the issue to you

**Comment by pawanharariya** [comment]
@clarlars  I searched through login of few sites and found out that this has two different interpretations: one that you suggested and second one where the icon implies the state you want to switch to and not the current state.

For example: sites like grammerly, mailchimp, instagram,etc use this implementation.
![Screenshot (81)](https://user-images.githubusercontent.com/43620548/115106647-abfdc800-9f83-11eb-9cca-13afcaef8d75.png)
![Screenshot (82)](https://user-images.githubusercontent.com/43620548/115106651-ae602200-9f83-11eb-9e01-92c849302870.png)

However facebook uses the implementation that you suggested.
![Screenshot (83)](https://user-images.githubusercontent.com/43620548/115106652-aef8b880-9f83-11eb-86c3-19c2c6af8ddf.png)

If we want to go by the same implementation then there are two other screens as well that require the change:
1. Admin password creation screen
2. Admin password challenge screen
- server password screen ( already mentioned above)

**Comment by sarahelshabrawy** [comment]
@pawanharariya I thought of the same thing as well. It's somehow confusing to be honest, however, I think that the implementation @clarlars suggested sounds more logical although I'm okay with both.

**Comment by linl33** [comment]
I think we should use the default Google provided. I've seen seen this implemented both ways. But the default that comes with the libraries is fine. So if the visibility icon was modified, IMO, we should restore to the default but if this is the stock settings I think it's fine.

**Comment by clarlars** [comment]
This issue stood out to me as it deviates from the previous behavior, but in the end, I am also fine with the default behavior provided by Google if no one has any objections.

**Comment by sarahelshabrawy** [comment]
That seems sensible, I'm fine with it as well, Just let me know if we're going to close this issue before I start working on it.

**Comment by wbrunette** [comment]
@sarahelshabrawy what is the google default? Please verify that we are following the Android standards. Once you do that we can close this.

**Comment by sarahelshabrawy** [comment]
Yes, I understood. Just wanted to make sure that everyone is okay with it.. I will go ahead then

**Comment by linl33** [comment]
The example shown on Google's Material Design website (https://material.io/components/text-fields/android#using-text-fields) implements the password visibility toggle as implemented by @pawanharariya.

@sarahelshabrawy I see that you've opened a pull request to invert the icons, did you find a source that recommends differently?

**Comment by pawanharariya** [comment]
@linl33 Yes the current implementation is the default one. However it was implemented by @VarunT11 and not me. I just found this issue interesting and was sharing my views.

**Comment by sarahelshabrawy** [comment]
@linl33 As @pawanharariya mentioned in his [comment](https://github.com/odk-x/tool-suite-X/issues/257#issuecomment-821788573) the default behaviour provided by material design is not implemented by all sites, like facebook .. Both implementations are okay, even material design used to implement the suggested implementation in previous versions. But as mentioned it was agreed to apply Android standards.
Just let me know please if I misunderstood anything.
